views/index.js

```diff
@@ -0,0 +1,32 @@
+fetch("http://news-summary-api.herokuapp.com/guardian?apiRequestUrl=http://content.guardianapis.com/search?show-fields=thumbnail&show-elements=image")
+  .then(response => {
+    console.log("response:")
+    console.log(response)
+    return response.json()
+  })
+  .then(data => {
+    console.log("data:")
+    console.log(data)
+    let subHeading = document.getElementById("subHeading")
+    data.response.results.forEach((article, index) => {
+      subHeading.insertAdjacentHTML('afterend', `<br>`)
+      subHeading.insertAdjacentHTML('afterend', `<img src="${data.response.results[index].fields.thumbnail}"><br>`)
+      subHeading.insertAdjacentHTML('afterend', `<a href="#${index}">${data.response.results[index].webTitle}</a><br>`)
+    })
+    return data;
+  })
+  .then(data => {
+    window.addEventListener("hashchange", function() {
```

Comment:
Your hashchange works really well. Mine makes the page jump around really inconveniently, and I couldn't work out why, did you come across that and if so what did you do to stop it? (From a user perspective, it might be good if the page jumps back up to the bit where the clicked on article appears)